ci/Makefile

```diff
@@ -1,11 +1,36 @@
-.PHONY: run restart-proxy restart-batch-proxy restart-all-proxies
+.PHONY: check-vars run restart-proxy restart-batch-proxy restart-all-proxies
 .PHONY: setup-conda-env build-hail-ci push-hail-ci test-locally
 .PHONY: run-service deploy
 
-PROJECT = $(shell gcloud config get-value project)
+PROJECT := $(shell gcloud config get-value project)
+
+# Grabs details of first instance matching HAIL_CI_INSTANCE
+DETAILS := $(shell gcloud compute instances list --format='table(NAME,STATUS,tags.list(),EXTERNAL_IP,ZONE)' | tail -n +2 | grep "$(HAIL_CI_INSTANCE)" --color=never | grep RUNNING | head -n 1 | tr -s ' ' '\n' )
+INSTANCE := $(word 1, $(DETAILS))
+EXTERNAL_IP := $(word 4, $(DETAILS))
+ZONE := $(word 5, $(DETAILS))
 
 HAIL_CI_LOCAL_BATCH_PORT ?= 8888
 
+# Must not be indented for conditions to work
+check-vars:
+ifeq ($(strip $(HAIL_CI_INSTANCE)),)
+		$(error HAIL_CI_INSTANCE variable not set or empty (HAIL_CI_INSTANCE='tag_name or instance_name'))
+endif
+
+ifeq ($(strip $(INSTANCE)),)
+		$(error Couldn't find an instance name matching $(HAIL_CI_INSTANCE))
```

Comment:
Ah I see now. What's the use case for referencing an instance by it's tag?

Reply:
You need to identify the instance somehow. Previously it was done by tag, hard coded as dk-test. This simply generalized that, and double checks that you actually have said instance to provide a less confusing error message.